src/Controls/src/Core/Handlers/Items2/iOS/TemplatedCell2.cs

```diff
@@ -76,26 +85,34 @@ public override UICollectionViewLayoutAttributes PreferredLayoutAttributesFittin
 
 			if (PlatformHandler?.VirtualView is not null)
 			{
-				if (ScrollDirection == UICollectionViewScrollDirection.Vertical)
-				{
-					var measure =
-						PlatformHandler.VirtualView.Measure(preferredAttributes.Size.Width, double.PositiveInfinity);
+				var constraints = ScrollDirection == UICollectionViewScrollDirection.Vertical
+					? new Size(preferredAttributes.Size.Width, double.PositiveInfinity)
+					: new Size(double.PositiveInfinity, preferredAttributes.Size.Height);
 
-					preferredAttributes.Frame =
-						new CGRect(preferredAttributes.Frame.X, preferredAttributes.Frame.Y,
-							preferredAttributes.Frame.Width, measure.Height);
-				}
-				else
+				if (_measureInvalidated || _cachedConstraints != constraints)
 				{
-					var measure =
-						PlatformHandler.VirtualView.Measure(double.PositiveInfinity, preferredAttributes.Size.Height);
-
-					preferredAttributes.Frame =
-						new CGRect(preferredAttributes.Frame.X, preferredAttributes.Frame.Y,
-							measure.Width, preferredAttributes.Frame.Height);
+					var measure = PlatformHandler.VirtualView.Measure(constraints.Width, constraints.Height);
+					_cachedConstraints = constraints;
+					_measuredSize = measure;
 				}
 
+				var size = ScrollDirection == UICollectionViewScrollDirection.Vertical
+					? new Size(preferredAttributes.Size.Width, _measuredSize.Height)
+					: new Size(_measuredSize.Width, preferredAttributes.Size.Height);
+
+				preferredAttributes.Frame = new CGRect(preferredAttributes.Frame.Location, size);
 				preferredAttributes.ZIndex = 2;
+
+				// We need to set the Frame on the virtual view.
+				// Subviews will eventually be arranged via LayoutSubviews.
+				var frame = new Rect(Point.Zero, size);
+				var virtualView = PlatformHandler.VirtualView!;
+				if (virtualView.Frame != frame)
+				{
+					PlatformHandler.VirtualView!.Arrange(frame);
```

Comment:
Unfortunately this is gonna trigger additional (unneeded) layouts when the `VirtualView`is a legacy layout.
Without this, transforms (like `Scale`) won't work.
Anyway the user can decide to use new layouts like `VerticalStackLayout` instead of `ContentView` as root node.